core/src/main/java/hudson/WebAppMain.java

```diff
@@ -396,8 +396,10 @@ public void contextDestroyed(ServletContextEvent event) {
         if(instance!=null)
             instance.cleanUp();
         Thread t = initThread;
-        if (t!=null)
+        if (t != null && t.isAlive()) {
+            LOGGER.log(Level.INFO, "Shutting down a Jenkins instance that was still starting up", new Throwable("reason"));
```

Comment:
I'd agree with @amuniz as well. If you feel this case is important enough, promote it to WARNING